### PR TITLE
add guide on how to install libgit2

### LINKED ISSUE #13732: Update function-controller documentation with how to build/install libgit2 on macOS
Add a section for a developers on how to build/install `libgit2` on macOS.

# How to build and install `libgit2` binary, required by function-controller, on macOS

1. Navigate to function controller root directory and verify the version of the git2go 

```
$ cat go.mod | grep git2go
        github.com/libgit2/git2go/v31 v31.4.14
```

2. Visit https://github.com/libgit2/git2go#git2go to check which version of `libgit2` you need to use (e.g for `libgit2` in version `1.1` the `git2go` version will be `v31`).

3. Clone libgit2 repository

```
git clone https://github.com/libgit2/libgit2.git
```

3. Checkout sources (the example below will checkout sources for `git2go` in version `v31`)

```
git checkout v1.1.0
```

4. Build and install

```
cmake -DCMAKE_OSX_ARCHITECTURES="x86_64" .
make install
```

### PR DESCRIPTION
**Pull request by grego952**
**Description**

Changes proposed in this pull request:

- add troubleshooting guide on how to build and install `libgit2` binary

**Related issue(s)**
Resolves #13732

### CODE REVIEW

**Review by kwiatekus — APPROVED** [review]

File: components/function-controller/README.md [hunk 1]
```diff
@@ -66,3 +66,31 @@ To develop the component, use the formulae declared in the [generic](/common/mak
 | **WEBHOOK_DEFAULTING_MINREPLICAS**        | Value of the minReplicas which webhook should set if origin equals null                   | `1`                  |
 | **WEBHOOK_DEFAULTING_MAXREPLICAS**        | Value of the maxReplicas which webhook should set if origin equals null                   | `1`                  |
 | **WEBHOOK_DEFAULTING_RUNTIME**            | Value of the runtime which webhook should set if origin equals null                       | `nodejs14`           |
+
+## Troubleshooting
+
+### How to build and install `libgit2` binary, required by the Function Controller, on macOS
```
Comment (klaudiagrz) [inline components/function-controller/README.md:72]: Please follow the [troubleshooting template](https://github.com/kyma-project/community/blob/main/templates/resources/troubleshooting.md).

File: components/function-controller/README.md [hunk 2]
```diff
@@ -66,3 +66,31 @@ To develop the component, use the formulae declared in the [generic](/common/mak
...
+
+1. Navigate to the Function Controller's root directory and verify the version of `git2go`:
+
+   ```bash
+   $ cat go.mod | grep git2go
+        github.com/libgit2/git2go/v31 v31.4.14
```
Comment (klaudiagrz) [inline components/function-controller/README.md:78]: Is the indentation intentional?

File: components/function-controller/README.md [hunk 3]
```diff
@@ -66,3 +66,31 @@ To develop the component, use the formulae declared in the [generic](/common/mak
...
+   ```
+2. Go to the git2go page to check which version of `libgit2` you must use (e.g. for `libigit2` in version 1.1 the `git2go` version is v31).
```
Comment (klaudiagrz) [inline components/function-controller/README.md:80]: A link to the page would be very useful here.

Comment (klaudiagrz) [inline components/function-controller/README.md:80]:
```suggestion
2. Go to the git2go page to check which version of libgit2 you must use. For example, for libigit2 in version 1.1, use git2go v31.
```

Comment (klaudiagrz) [reply components/function-controller/README.md:80]: Why the code formatting?

File: components/function-controller/README.md [hunk 4]
```diff
@@ -66,3 +66,31 @@ To develop the component, use the formulae declared in the [generic](/common/mak
...
+   
+3. Clone `libgit2` repository:
```
Comment (klaudiagrz) [inline components/function-controller/README.md:82]:
```suggestion
3. Clone the `libgit2` repository:
```

File: components/function-controller/README.md [hunk 5]
```diff
@@ -66,3 +66,31 @@ To develop the component, use the formulae declared in the [generic](/common/mak
...
+
+   ```bash
+   git clone https://github.com/libgit2/libgit2.git
+   ```
+4. Checkout sources (the following example is for `git2go` version v31):
```
Comment (klaudiagrz) [inline components/function-controller/README.md:87]:
```suggestion
4. Check out the sources. In this example, the sources are for git2go v31:
```

File: components/function-controller/README.md (hunk 2 above)
Comment (grego952) [reply components/function-controller/README.md:78]: I guess. It was like that in the [source issue](https://github.com/kyma-project/kyma/issues/13732).

File: components/function-controller/README.md [hunk 6]
```diff
@@ -66,3 +66,45 @@ To develop the component, use the formulae declared in the [generic](/common/mak
...
+
+   ```bash
+   $ cat go.mod | grep git2go
+   ```
+   You should get a result similar to this example:
+
+   ```bash
+   github.com/libgit2/git2go/v31 v31.4.14
+   ```
+2. Go to the [git2go page](https://github.com/libgit2/git2go#git2go) to check which version of libgit2 you must use. For example, for libigit2 in version 1.1, use git2go v31.
```
Comment (klaudiagrz) [inline components/function-controller/README.md:94]:
```suggestion
2. Go to the [git2go page](https://github.com/libgit2/git2go#git2go) to check which version of libgit2 you must use. For example, for git2go v31, use libigit2 in version 1.1.
```

Comment (klaudiagrz) [reply components/function-controller/README.md:94]: Shouldn't we adjust the libigit2 version to the git2go version we have?

File: components/function-controller/README.md [hunk 7]
```diff
@@ -66,3 +66,45 @@ To develop the component, use the formulae declared in the [generic](/common/mak
...
+3. Clone the `libgit2` repository:
+
+   ```bash
+   git clone https://github.com/libgit2/libgit2.git
+   ```
+4. Check out the sources. In this example, the sources are for git2go v31:
+   ```bash
+   git checkout v1.1.0
+   ```
+5. Build and install:
```
Comment (klaudiagrz) [inline components/function-controller/README.md:105]:
```suggestion
5. Build and install the libgit2 binary:
```

File: components/function-controller/README.md [hunk 8]
```diff
@@ -66,3 +66,47 @@ To develop the component, use the formulae declared in the [generic](/common/mak
...
+
+
+### Symptom
+
+Function Controller tests keep failing with such an error message:
+`error: Invalid libgit2 version; this git2go supports libgit2 between vA.B.C and vX.Y.Z`
+
+### Cause
+
+Function Controller tests failing due to the wrong version of the libgit2 binary. The required version of the binary is 1.1.
```
Comment (klaudiagrz) [inline components/function-controller/README.md:80]:
```suggestion
Function Controller tests are failing due to the wrong version of the libgit2 binary. The required version of the binary is 1.1.
```

File: components/function-controller/README.md [hunk 9]
```diff
@@ -66,3 +66,47 @@ To develop the component, use the formulae declared in the [generic](/common/mak
...
+
+### Remedy
+
+Build and Install the libgit2 binary required by the Function Controller on macOS. Follow these steps:
```
Comment (klaudiagrz) [inline components/function-controller/README.md:84]:
```suggestion
Build and install the libgit2 binary required by the Function Controller on macOS. Follow these steps:
```

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**